### PR TITLE
Deck builder refactor part2

### PR DESCRIPTION
**Pull request by lily-pytel**
Sorry for the bloated PR!

Part 2 of refactoring.

Deck builder page:

image

Card page:

image

I'm not explicitly working on the card search page, it just uses a lot of the same things

### CODE REVIEW

File: next/components/card-search-filters.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          margin-right: 35px;
+          margin-bottom: 35px;
+        }
+        hr {
+          margin-bottom: 10px;
+          margin-left: 0;
+          width: 95%;
+          border: 0;
+          height: 1px;
+          background-image: linear-gradient(
```
Comment (jtrussell) [inline next/components/card-search-filters.js]: Those dang gradient hrs. I'm cool with these colors being explicit, just something to keep in mind if ever get around to actually supporting multiple themes.

Comment (lily-pytel) [reply next/components/card-search-filters.js]: Nah, gross, I'll change in part 4

**Comment by alexmiddeleer** [comment]
noice

File: next/components/card-search-form.js [hunk 2]
```diff
@@ -33,25 +30,31 @@ export default function CardSearchForm(props) {
...
-      Health/Durability: <DefenseFilter onChange={setDefenses} />
-      <br />
-      <br />
       <FactionFilters
         onFactionClick={newFactions => setFactions(newFactions)}
       />
+      <CardSearchFilters
+        manaCosts={manaCosts}
+        strengths={strengths}
+        healths={defenses}
```
Comment (alexmiddeleer) [inline next/components/card-search-form.js]: ideally we don't switch the variable name here

File: next/components/card-search-form.js [hunk 3]
```diff
@@ -33,25 +30,31 @@ export default function CardSearchForm(props) {
...
+        rarities={rarities}
+        types={supertypes}
+        setCardManaCosts={setManaCosts}
+        setCardStrengths={setStrengths}
+        setCardHealths={setDefenses}
```
Comment (alexmiddeleer) [inline next/components/card-search-form.js]: dittos

File: next/components/card-search-form.js [hunk 4]
```diff
@@ -33,25 +30,31 @@ export default function CardSearchForm(props) {
...
-      <br />
       <FactionFilters
         onFactionClick={newFactions => setFactions(newFactions)}
       />
+      <CardSearchFilters
+        manaCosts={manaCosts}
+        strengths={strengths}
+        healths={defenses}
+        rarities={rarities}
+        types={supertypes}
```
Comment (alexmiddeleer) [inline next/components/card-search-form.js]: dittos

**Pull request merged**